librad/src/git/storage.rs

```diff
@@ -546,6 +563,55 @@ impl Storage {
         })
     }
 
+    /// Get the [`Project`] at the provided [`RadUrn`]
+    ///
+    /// # Errors
+    ///
+    /// * If the `RadUrn` does not point to a project.
+    pub fn project(&self, urn: &RadUrn) -> Result<Project<Draft>, Error> {
+        self.entity_info_or(&urn, |info| match info {
+            EntityInfo::Project(project_info) => Some(project_info),
+            _ => None,
+        })
+    }
+
+    /// Get the [`User`] at the provided [`RadUrn`]
+    ///
+    /// # Errors
+    ///
+    /// * If the `RadUrn` does not point to a user.
+    pub fn user(&self, urn: &RadUrn) -> Result<User<Draft>, Error> {
+        self.entity_info_or(&urn, |info| match info {
+            EntityInfo::User(user_info) => Some(user_info),
+            _ => None,
+        })
+    }
+
+    /// Get the [`Entity`] found at the provided [`RadUrn`]. Then try to map the
+    /// [`EntityInfo`] into the target `T`.
+    ///
+    /// # Errors
+    ///   * If the entity resolution fails.
+    ///   * If the mapping fails, an [`Error::IncorrectEntityInfo`] will be
+    ///     returned.
+    pub fn entity_info_or<F, T>(
```

Comment:
I understand your desire to make the error message informative, but I don't think it's worth it -- the entity "kind" is not exhaustive. Iow, why not simply try to deserialize the blob contents into `T`, and if that fails return either the raw bytes or the `serde_json::Value` in the error payload?

Reply:
Ya, fair point. I'll make the change.

Reply:
Makes much more sense :ok_hand: The irony being that this branch is called specific-entity.